library/common/src/main/java/com/google/android/exoplayer2/video/HevcConfig.java

```diff
@@ -72,20 +79,47 @@ public static HevcConfig parse(ParsableByteArray data) throws ParserException {
           System.arraycopy(
               data.getData(), data.getPosition(), buffer, bufferPosition, nalUnitLength);
           bufferPosition += nalUnitLength;
+
+          if (nalUnitType == SPS_NUT && j == 0) {
+            codecs = buildCodecString(data, nalUnitLength);
+          }
+
           data.skipBytes(nalUnitLength);
         }
       }
 
-      List<byte[]> initializationData = csdLength == 0 ? null : Collections.singletonList(buffer);
-      return new HevcConfig(initializationData, lengthSizeMinusOne + 1);
+      @Nullable List<byte[]> initializationData = csdLength == 0 ? null : Collections.singletonList(buffer);
+      return new HevcConfig(initializationData, lengthSizeMinusOne + 1, codecs);
     } catch (ArrayIndexOutOfBoundsException e) {
       throw new ParserException("Error parsing HEVC config", e);
     }
   }
 
-  private HevcConfig(@Nullable List<byte[]> initializationData, int nalUnitLengthFieldLength) {
+  private static String buildCodecString(ParsableByteArray data, int nalUnitLength) {
+    final ParsableNalUnitBitArray bitArray = new ParsableNalUnitBitArray(data.getData(), data.getPosition(), data.getPosition() + nalUnitLength);
+    bitArray.skipBits(24);
+    final int generalProfileSpace = bitArray.readBits(2);
+    final boolean generalTierFlag = bitArray.readBit();
+    final int generalProfileIdc = bitArray.readBits(5);
+    final int generalProfileCompatibilityFlags = bitArray.readBits(32);
```

Comment:
@equeim I think the order of bits in the resulting int is reversed compared to what's needed to build the codecs string. Please could you confirm whether that's consistent with your understanding or if I'm missing something?

I will take care of making the change if it's needed when merging this, so please just let me know whether it is correct. Thanks!

Reply:
I'm not sure, actually. MPEG-DASH for DVB specification says that bits must be reversed, but MPEG-4 Part 15 (where codecs string for HEVC is defined officially) doesn't mention it. Although I don't have access to the latest version of MPEG-4 Part 15 yet so I can't check whether it was added later.
It seems that almost all HLS/DASH manifests out there have it reversed though, and that was google/shaka-packager does too (https://github.com/google/shaka-packager/blob/d90cf9a0fd87833f7d9210ffe65d175dcf994c42/packager/media/codecs/hevc_decoder_configuration_record.cc#L133).